enterprise/mqtt/src/test/java/io/crate/mqtt/operations/MqttIntegrationTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.mqtt.operations;
+
+import com.google.common.primitives.Ints;
+import io.crate.integrationtests.SQLTransportIntegrationTest;
+import io.crate.mqtt.netty.Client;
+import io.crate.settings.SharedSettings;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.PortsRange;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import static io.crate.mqtt.netty.Netty4MqttServerTransport.MQTT_ENABLED_SETTING;
+import static io.crate.mqtt.netty.Netty4MqttServerTransport.MQTT_PORT_SETTING;
+
+public abstract class MqttIntegrationTest extends SQLTransportIntegrationTest {
+
+    private static final Logger LOGGER = Loggers.getLogger(MqttIntegrationTest.class);
+    private static final String PORT_RANGE = "11883-11983";
+    private static final Queue<Integer> AVAILABLE_PORTS =
+        new ConcurrentLinkedQueue<>(Ints.asList(new PortsRange(PORT_RANGE).ports()));
+    static Client MQTT_CLIENT;
+
+    @BeforeClass
+    public static void setUpMqttClient() {
+        MQTT_CLIENT = createMqttClient();
+    }
+
+    static Client createMqttClient() {
+        Integer port = AVAILABLE_PORTS.peek();
+        LOGGER.info("Creating local MQTT client on port {}", port);
+        return new Client("127.0.0.1", port);
+    }
+
+    @Before
+    public void connectMqttClient() throws Exception {
+        int attempts = 10;
+        while (attempts > 0) {
+            try {
+                MQTT_CLIENT.connect();
```

Comment:
I'm not sure if this approach is now safe. If someone now really executes the
tests in parallell, I think it could happen that a client connects to the mqtt
endpoint of a *different* "test launch".

To avoid this issue, it would be possible to retrieve the MqttTransport via
`internalCluster()` and get the port from the boundAddress (which has to be
made accessible, but I think that tradeoff is worth it).